fix(client/server): Radio Object Creation

ihyajb opened the pull request:
## Description

Now uses locally spawned client objects synced with state bags. Instead of fully server sided objects

Closes #42
Supersedes #43 

Tested with CL2 and all fine also 
![Screenshot 2025-03-06 074409](https://github.com/user-attachments/assets/174a3350-d27c-4fc7-9270-bad96981aee7)

<!-- What does your pull request change? Why should it be merged? Does it fix an issue? -->

## Checklist

<!-- Put an x inside the [ ] to check an item, like so: [x] -->

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

ihyajb committed:
fix(client/server): Radio Object Creation

Uses client side spawn and state bags now

Co-Authored-By: Ehbw <106408826+Ehbw@users.noreply.github.com>

Ehbw commented on server/main.lua:
```diff
@@ -24,57 +22,7 @@ for channel, jobs in pairs(restrictedChannels) do
     local src = source
-
-    deleteProp(src)
+    Player(src).state.isHoldingRadio = bool
```
It may be worth making sure that only a bool can be passed to this. As otherwise a client could sent a bunch of useless data to be attached to them

ihyajb replied:
Good Idea

FjamZoo reviewed (changes requested):
This is close, but we also need a way to handle whenever a player leaves the other player scope, the way to handle this is 1 either loop or use the onPlayerDrop client event refer to Renewed-Weaponscarry here to see how I've done it:

https://github.com/Renewed-Scripts/Renewed-Weaponscarry/blob/main/modules/carry.lua#L16

ihyajb commented:
Should be good now, renamed the `netId` var in the handler to be `serverId` to keep it more consistent also.

FjamZoo commented on client/main.lua:
```diff
@@ -85,35 +85,49 @@ local function toggleRadio(toggle)
+end)
+
+RegisterNetEvent('onPlayerDropped', function(serverId)
+    if not Radios[serverId] then return end
+    SetEntityAsMissionEntity(Radios[serverId], true, true)
+    DeleteEntity(Radios[serverId])
+    Radios[serverId] = nil
```
Looks good, the last little bit would probably just make this into a function so we adhere to the DRY principle.

Other than that this PR is good to go

ihyajb replied:
will do, also forgot to add the resource stop / restart check so will add that back also

ihyajb commented:
- Added a `waitFor` to make sure the ped is valid
- Removed no longer needed locales
- Moved the cleanup to a function
- Added cleanup for resource stops / restarts

FjamZoo reviewed (approved):
Beutiful thanks AJ, merge whenever :)
